File: 55.jump game/solve.cpp
#include <vector>
using namespace std;
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        vector<bool> dp(n);
        dp[n-1] = true;
        for(int i=n-2;i>=0;i--){
            for (int j = i+1; j<=i+nums[i] && j<n; j++)
            {
                if (dp[j]==true){
                    dp[i]=true;
                    break;
                }
            }
        }
        return dp[0];
    }
};

File: 55.jump game/solve.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n=len(nums)
        dp = [False]*n
        dp[n-1] = True
        for i in range(n-2, -1, -1):
            for detect in range(i+1, min(n,i+nums[i]+1)):
                if dp[detect]:
                    dp[i]=True
                    break
        return dp[0]